datacontract/export/dbt_converter.py

```diff
@@ -130,25 +132,25 @@ def _supports_constraints(model_type):
     return model_type == "table" or model_type == "incremental"
 
 
-def _to_columns(fields: Dict[str, Field], supports_constraints: bool, supports_datatype: bool) -> list:
```

Comment:
The dbt supported by datacontract-cli is 1.8 or higher, and dbt has supported `data_type` since 1.5. Therefore, we decided that it would be fine to always set `supports_datatype` to True, and removed it from the arguments.

- https://github.com/datacontract/datacontract-cli/blob/main/pyproject.toml#L89-L91
- https://github.com/dbt-labs/dbt-core/blob/1.5.latest/CHANGELOG.md

Reply:
It's not about the dbt version. 

We use the same code to generate the YAML for sources and for models. Sources do not support the datatype attribute, only models do. Can you revert, please?

Reply:
@simonharrer Thank you for your reply!

> Sources do not support the datatype attribute, only models do

I tried reading the source code for dbt-core. The definition for Sources is done in `SourceDefinition`, and the definition for columns is done in `ColumnInfo`. `data_type` is included in this.

- https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/artifacts/resources/v1/source_definition.py
- https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/artifacts/resources/v1/components.py#L65

In JSON Schema for the dbt yml file, `data_type` is also defined in `column_properties` within `columns`.

- https://github.com/dbt-labs/dbt-jsonschema/blob/main/schemas/latest/dbt_yml_files-latest.json#L903-L908
- https://github.com/dbt-labs/dbt-jsonschema/blob/main/schemas/latest/dbt_yml_files-latest.json#L1208-L1210

In fact, if you look at the contents generated by `manifest.json`, you can see that `data_type` is defined in Sources, so you can tell that Sources supports the `data_type` attribute.

```json
{
  "metadata": {
    "dbt_schema_version": "https://schemas.getdbt.com/dbt/manifest/v12.json",
    "dbt_version": "1.8.8",
    "generated_at": "2024-12-05T23:58:59.230157Z",
    "invocation_id": "41248e0c-2224-4a15-9c47-8b69f800bf76",
    "env": {},
    "project_name": "my_project",
    "project_id": "1ae1414bfc8e33f4f3905e27dbd8c461",
    "user_id": null,
    "send_anonymous_usage_stats": true,
    "adapter_type": "bigquery"
  },
  "nodes": {
    "source.my_project.my_dataset.my_table": {
      "database": "my_project",
      "schema": "my_dataset",
      "name": "my_table",
      "resource_type": "source",
      "package_name": "my_package",
      ...,
      "source_name": "my_project",
      "source_description": "",
      "loader": "",
      "identifier": "my_table",
      "quoting": {
        "database": null,
        "schema": null,
        "identifier": null,
        "column": null
      },
      "loaded_at_field": null,
      "freshness": null,
      "external": null,
      "description": "",
      "columns": {
        "shopId": {
          "name": "shopId",
          "description": "shop id",
          "meta": {},
          "data_type": "STRING",
          "constraints": [],
          "quote": null,
          "tags": []
        },
        "userId": {
          "name": "userId",
          "description": "user id",
          "meta": {},
          "data_type": "STRING",
          "constraints": [],
          "quote": null,
          "tags": []
        }
      },
      "meta": {
        "manual_data_usage": false
      },
      "source_meta": {},
      "tags": [],
      "config": {
        "enabled": true
      },
      "patch_path": null,
      "unrendered_config": {},
      ...,
      "created_at": 1731295492.503308
    }
  }
}
```

From the above survey, I don't think it is necessary to revert the relevant line, but what do you think, @simonharrer?

Reply:
So, does this mean that the conclusion here is that `data_type` is also supported by Sources, so there is no need to revert the corresponding line?

If so, please continue to review it. If there is anything that concerns you, please feel free to point it out!